src/dbnode/storage/namespace/schema.go

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package namespace
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"io/ioutil"
+	"github.com/golang/protobuf/proto"
+	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/jhump/protoreflect/desc"
+	nsproto "github.com/m3db/m3/src/dbnode/generated/proto/namespace"
+	xerrors "github.com/m3db/m3x/errors"
+	"strings"
+)
+
+var (
+	errInvalidSchema        = errors.New("invalid schema definition")
+	errSchemaNotFound       = errors.New("schema is not found")
+	errSchemaRegistryEmpty  = errors.New("schema registry is empty")
+	errInvalidSchemaOptions = errors.New("invalid schema options")
+)
+
+type schemaDescr struct {
+	deployId string
+	md       *desc.MessageDescriptor
+}
+
+func (s *schemaDescr) DeployId() string {
+	return s.deployId
+}
+
+func (s *schemaDescr) Equal(o SchemaDescr) bool {
+	if s == nil && o == nil {
+		return true
+	}
+	if s != nil && o == nil || s == nil && o != nil {
+		return false
+	}
+	if _, ok := o.(*schemaDescr); !ok {
+		return false
+	}
+	return s.DeployId() == o.DeployId()
+}
+
+func (s *schemaDescr) Get() *desc.MessageDescriptor {
+	return s.md
+}
+
+func (s *schemaDescr) String() string {
+	if s.md == nil {
+		return ""
+	}
+	return s.md.String()
+}
+
+type schemaRegistry struct {
+	options  *nsproto.SchemaOptions
```

Comment:
As far as I can tell this is unused. I'd prefer not to carry this raw proto message around everywhere. Can we remove this field?

Reply:
Oh are you keeping around so we can convert back to the original proto?

Reply:
yes, it is for that purpose.